mobile/lib/infrastructure/repositories/sync_api.repository.dart

```diff
@@ -0,0 +1,112 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter/foundation.dart';
+import 'package:immich_mobile/domain/interfaces/sync_api.interface.dart';
+import 'package:immich_mobile/domain/models/sync/sync_event.model.dart';
+import 'package:immich_mobile/services/api.service.dart';
+import 'package:openapi/api.dart';
+import 'package:http/http.dart' as http;
+
+class SyncApiRepository implements ISyncApiRepository {
+  final ApiService _api;
+  const SyncApiRepository(this._api);
+
+  @override
+  Stream<List<SyncEvent>> watchUserSyncEvent() {
+    return _getSyncStream(
+      SyncStreamDto(types: [SyncRequestType.usersV1]),
+    );
+  }
+
+  @override
+  Future<void> ack(String data) {
+    return _api.syncApi.sendSyncAck(SyncAckSetDto(acks: [data]));
+  }
+
+  Stream<List<SyncEvent>> _getSyncStream(
+    SyncStreamDto dto, {
+    int batchSize = 5000,
+  }) async* {
+    final client = http.Client();
```

Comment:
We can use the client from the ApiService. This way, we can mock them and add tests for the repository as well.
```suggestion
    final client = _api.apiClient.client;
```

Reply:
This change triggers an error

```
I/flutter (10522): FlutterError - Catch all: ClientException: HTTP request failed. Client is already closed., uri=http://10.1.15.216:2283/api/sync/stream 
I/flutter (10522):  #0      IOClient.send (package:http/src/io_client.dart:110:7)
I/flutter (10522): #1      SyncApiRepository._getSyncStream (package:immich_mobile/infrastructure/repositories/sync_api.repository.dart:52:37)
I/flutter (10522): <asynchronous suspension>
I/flutter (10522): #2      SyncStreamService.syncUsers.<anonymous closure> (package:immich_mobile/domain/services/sync_stream.service.dart:16:56)
I/flutter (10522): <asynchronous suspension>
```

Reply:
Oh shoot! Yes, the client inside the service is reused for all connections, so closing it would trigger the error. We can probably resort to optionally injecting the client inside the method only for testing. i.e,

```dart
Stream<List<SyncEvent>> watchUserSyncEvent({http.Client? client}) {
  return _getSyncStream(
    SyncStreamDto(types: [SyncRequestType.usersV1]),
    client: client,
  );
}

 Stream<List<SyncEvent>> _getSyncStream(
  SyncStreamDto dto, {
  int batchSize = 5000,
  http.Client? client,
}) async* {
  client ??= http.Client();

```